user_sync/rules.py

```diff
@@ -502,6 +502,9 @@ def process_strays(self, umapi_connectors):
             self.write_stray_key_map()
         if self.will_manage_strays:
             max_missing = self.options['max_adobe_only_users']
+            if isinstance(max_missing, str) and '%' in max_missing:
+                percent = float(max_missing.strip('%'))/100
+                max_missing = int((self.primary_user_count - self.excluded_user_count) * percent)
```

Comment:
Since you're already parsing the percentage as an integer in the config (in `max_missing_percent`), can't you save that to the `options` dict and use that here?  If the `max_missing_percent` key is in `options` and doesn't contain a null value, use it to derive `max_missing`, otherwise use the value of `max_adobe_only_users`.  You can even convert it to a float in the config.